microsoft-edge/webview2/how-to/javascript.md

```diff
@@ -30,6 +30,24 @@ Use the following functions to begin embedding JavaScript in your WebView app.
 | [ExecuteScriptAsync](/dotnet/api/microsoft.web.webview2.wpf.webview2.executescriptasync) | Run JavaScript in a WebView control. For more information, navigate to the Get Started tutorial. |
 | [OnDocumentCreatedAsync](/microsoft-edge/webview2/reference/win32/icorewebview2#addscripttoexecuteondocumentcreated) | Runs when the Document Object Model (DOM) is created. |
 
+<!-- ====================================================================== -->
+## Scenario: ExecuteScript JSON Encoded Results
+
+Because the result of `ExecuteScriptAsync` is JSON encoded, if the result evaluating the JavaScript is a string you will receive a JSON encoded string and not the value of the string. For example, the following script results in a string with the following value including the quotes at the start and end, and the escaping slashes. 
+ * Script: "example \" \' \\ "; 
+ * Result: "example \" ' \\ "
+The script returns a string that `ExecuteScript` JSON encodes for you. If you call `JSON.stringify` from your script, then the result is doubly encoded as a JSON string the value of which is a JSON string.
+
+Only the properties in the result will be included in the JSON encoded object, not inherited properties. Most DOM objects inherit all properties, so you'll need to explicitly copy their values on to another object to return. For example:
+
+Script              | Result
+---                 | ---
+performance.memory  | {}
+(() => { const {totalJSHeapSize, usedJSHeapSize} = performance.memory; return {totalJSHeapSize, usedJSHeapSize}; })(); |  {"totalJSHeapSize":4434368,"usedJSHeapSize":2832912}
+
+When we return just `performance.memory` we don't see any of its properties in the result because all properties are inherited. If instead we copy particular property values from `performance.memory` into our own new object to return which properties are directly set, then we do see those properties in the result.
+
+When executing script via `ExecuteScriptAsync` that script is run in the global context. It helps to have your script in an anonymous function so that any variables you define aren't polluting the global context. For example, if you run the script `const example = 10;` more than once, the subsequent times you run the script will throw an exception because example was defined the first time you ran it. If you instead run the script `(() => { const example = 10; })();` the example variable is defined in the context of that anonymous function. That way it is not polluting the global context and can be run more than once.
```

Comment:
```suggestion
When executing script via `ExecuteScriptAsync`, that script is run in the global context. It helps to have your script in an anonymous function, so that any variables you define aren't polluting the global context. For example, if you run the script `const example = 10;` more than once, the subsequent times you run the script will throw an exception, because `example` was already defined the first time you ran it. If you instead run the script `(() => { const example = 10; })();`, the `example` variable is defined in the context of that anonymous function. That way, the variable definition isn't polluting the global context, and so the line can be run more than once.
```

Reply:
done